File: jsParser/parser/Logical_termVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Logical_term = require('../domain/Logical_term');

const Logical_factorVisitor = require('./Logical_factorVisitor');


class Logical_termVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitLogical_term(ctx) {
        var logical_factor_list = [];

        if (ctx.logical_factor()) {    
            var logical_factorVisitor = new Logical_factorVisitor.Logical_factorVisitor();
            ctx.logical_factor().forEach(factor=> {
                logical_factor_list.push(logical_factorVisitor.visitLogical_factor(factor));
            });
        }
        
        return new Logical_term.Logical_term(logical_factor_list);
    }
};

Logical_termVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitLogical_term = this.visitLogical_term;
exports.Logical_termVisitor = Logical_termVisitor;

File: java/parser/src/main/java/gov/lbl/parser/parser/Logical_termVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Logical_factor;
import gov.lbl.parser.domain.Logical_term;

import static java.util.stream.Collectors.toList;

public class Logical_termVisitor extends modelicaBaseVisitor<Logical_term> {
    @Override
    public Logical_term visitLogical_term(modelicaParser.Logical_termContext ctx) {
        Logical_factorVisitor logical_factorVisitor = new Logical_factorVisitor();
        List<Logical_factor> logical_factor_list = ctx.logical_factor()
                .stream()
                .map(logical_factor -> logical_factor.accept(logical_factorVisitor))
                .collect(toList());
        return new Logical_term(logical_factor_list);
    }
}